nncf/experimental/post_training/graph/model_transformer.py

```diff
@@ -0,0 +1,129 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from abc import ABC
+from abc import abstractmethod
+from typing import List
+from typing import TypeVar
+
+from nncf.common.graph.model_transformer import ModelTransformer
+from nncf.common.graph.transformations.commands import TransformationCommand
+from nncf.common.graph.transformations.layout import TransformationLayout
+from nncf.experimental.post_training.statistics.statistic_point import StatisticPointsContainer
+
+ModelType = TypeVar('ModelType')
+
+
+# pylint: disable=no-member
+class StaticModelTransformerBase(ModelTransformer, ABC):
+    QUANTIZER_NAME_PREFIX = 'QuantizeLinear_'
+    DEQUANTIZER_NAME_PREFIX = 'DequantizeLinear_'
+    SCALE_TENSOR_NAME_PREFIX = 'scale_'
+    ZERO_POINT_NAME_PREFIX = 'zero_point_'
+
+    def __init__(self, model: ModelType):
+        super().__init__(model)
+        self._transformation_layout = None
+        self._transformations_list = []
+
+        # Transformation commands list
+        self._output_insertion_command = None
```

Comment:
basically you can use here common Transformations and you don't need to override them in interstated classes.

Reply:
Can you please explain how we can use common Transformations here, in the Base class?

Reply:
for example we have ONNXOutputInsertionCommand and OpenVINOOutputInsertionCommand which are inherited from OutputInsertionCommand. 
you can check whether the command is instance of OutputInsertionCommand and calls corresponding function which is implemented in backend-specific class.

Reply:
`isinstance(...)` checks should be generally avoided in a good OOP architecture. It is always possible to utilize polymorphism instead. The only good use of `isinstance(...)` that I could think of in Python is the situation where we have to apply various, subclass-specific operations to a collection of base class object references. Usually one would have to implement a visitor pattern for this, but `isinstance` won't be adding too much overhead for the already slow Python and would be shorter.

I'm not sure that this is the place where we would profit from a visitor pattern, and, by extension - from `isinstance`.